### DISCUSSION TITLE
PTZ Cameras get out of sync with NDI plugin on OBS

### DISCUSSION THREAD

**Discussion by pl22740** [DistroAV/DistroAV#947] [General]
We use two PTZ N30X cameras for our worship stream.  SInce upgrading to OBS 29 and up, one or the other camera will get out of sync with the real time video.  It seems to happen at random and I can't find a way to force it to happen.  I escalated the problem to PTZOptics and they see the same behavior in their lab with 2 cameras.  They don't see it with any other video software.  We don't see it with SPlit-Cam as well.  Has anybody else seen this behavior?  Any idea on how to fix?  Our system is a more than adequate Dell PC with NVIDIA GPU connected to the cameras with a 1Gb switch.  I can fix it in real time by unplugging the cameras from the switch and plugging back in (PoE switch so that cycles power on the cameras).  Any advice appreciated.  Glad to send logs but we don't see anything in the log that points to a problem.

**Comment by MarximusMaximus** [comment]
FWIW, I have similar problems in general with NDI lately (i.e. the last six+ months). I'm even using OBS 27 and the old plugin too. I /feel like/ that this is like a networking issue, maybe a QOS (Quality of Service) config  issue. That or it's a long standing very edge case hard to find in the code bug. Could be OBS itself even.

I've had it when target machines are Macs or PCs. I've had it when sources are OBS-ndi on PC, OBS-ndi on mac, NDI Capture HX on PC, NDI Camera HX on iOS. I've had it whether over wifi or ethernet.

**Comment by pl22740** [comment]
We've been chasing this issue for over a year I think.  I'm surprised that this hasn't been made visible by many others and investigated.  We're going to punt on NDI and try using capture cards.  THanks for the response.
